CHANGELOG.md

```diff
@@ -3,6 +3,344 @@
+* `EventCollector` is now deprecated, as its use usually involved anti-patterns around fallibility. However, its functionality can still be replicated using `collector::collect`. See example 10 for more details.
+* The `RelatedId` and `RelatedIdsForEventType` types have been removed as they were only used by `EventCollector`. Methods for retrieving them from events have also been removed; if users wish to extract "related" ids from an event, they should do so directly from the event's fields. The removed methods are the following:
+    - `Event::user_id`
+    - `Event::guild_id`
+    - `Event::channel_id`
+    - `Event::message_id`
+    - `EventType::related_ids`
+
+### Commands
+
+Serenity now uses simply `command` instead of `application_command` in all places, except for the `Permissions::USE_APPLICATION_COMMANDS` constant. This includes methods on the `Http`, `GuildId`, `Guild`, `PartialGuild`, and `Command` types, as well as a few miscellaneous places:
+
+| serenity v0.11 | serenity v0.12 |
+| --- | --- |
+| `Http::*_{global,guild}_application_command*` | `Http::*_{global,guild}_command*` |
+| `{GuildId,Guild,PartialGuild}::*_application_command*` | `{GuildId,Guild,PartialGuild}::*_command*` |
+| `Command::*_global_application_command*` | `Command::*_global_command*` |
+| `Interaction::application_command` | `Interaction::command` |
+| `EventHandler::application_command_permissions_update` | `EventHandler::command_permissions_update` |
+| `Route::ApplicationCommand*` | `Route::Command*` |
+| `Permissions::use_application_commands` | `Permissions::use_commands` |
+
+Additionally, the following command types have been renamed:
+| serenity v0.11 | serenity v0.12 |
+| --- | --- |
+| `CreateApplicationCommand` | `CreateCommand` |
+| `CreateApplicationCommandOption` | `CreateCommandOption` |
+| `CreateApplicationCommandPermissionData` | `CreateCommandPermission` |
+| `CreateApplicationCommandPermissionsData` | `EditCommandPermissions` |
+| `CommandPermission` | `CommandPermissions` |
+| `CommandPermissionData` | `CommandPermission` |
+
+Also, the methods on `CreateCommandPermission`, such as `new`, `kind`, etc. have been replaced with constructors for each type of permission, e.g. `role`, `user`, `channel`, etc., to avoid a possible mismatch between `kind` and the id that gets passed in.
+
+Finally, the `{GuildId,Guild,PartialGuild}::create_command_permission` method has been renamed to `edit_command_permission` to more accurately reflect its behavior.
+
+### Cache
+
+* Cache methods now (mostly) return a `CacheRef` type that wraps a reference into the cache. Other methods that returned a map, now return a wrapper type around a reference to the map, with a limited API to prevent accidental deadlocks. This all helps reduce the number of clones when querying the cache. Those wishing to replicate the old behavior can simply clone the references wholesale to obtain the wrapped data.
```

Comment:
```suggestion
* Cache methods now (mostly) return a `CacheRef` type that wraps a reference into the cache. Other methods that returned a map, now return a wrapper type around a reference to the map, with a limited API to prevent accidental deadlocks. This all helps reduce the number of clones when querying the cache. Those wishing to replicate the old behavior can simply call `.clone()` on the wrapper type to obtain the wrapped data.
```

Reply:
imo "return type" flows a bit better than "wrapper type"